dyson/solvers/chempot.py

```diff
@@ -120,6 +120,56 @@ def get_greens_function(self):
         return self.gf.copy(chempot=self.chempot, deep=False)
 
 
+class AufbauPrincipleBisect(AufbauPrinciple):
+
+    def _kernel(self):
+        energies = self.gf.energies
+        couplings_l, couplings_r = self.gf._unpack_couplings()
+        weights = self.gf.weights()
```

Comment:
maybe this is a stupid question, but `self.gf.weight()` is already O(nmo * naux), so is bisection only faster by virtue of vectorisation in calculating the weights? In the non-bisected algo you need O(nocc * naux) to calculate the weights but they're done one at a time so i can see how it would be less efficient in practice